testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -247,125 +255,176 @@ def check_DensityAnalysis(self, ag, ref_meandensity,
-    def test_run(self, mode, universe, tmpdir):
+    def test_run(self, mode, universe, tmpdir, client_DensityAnalysis):
         updating = (mode == "dynamic")
         self.check_DensityAnalysis(
             universe.select_atoms(self.selections[mode], updating=updating),
             self.references[mode]['meandensity'],
-            tmpdir=tmpdir
+            tmpdir=tmpdir,
+            client_DensityAnalysis=client_DensityAnalysis,
         )
 
-    def test_sliced(self, universe, tmpdir):
+    def test_sliced(self, universe, tmpdir, client_DensityAnalysis):
         self.check_DensityAnalysis(
             universe.select_atoms(self.selections['static']),
             self.references['static_sliced']['meandensity'],
             tmpdir=tmpdir,
+            client_DensityAnalysis=client_DensityAnalysis,
             runargs=dict(start=1, stop=-1, step=2),
         )
 
-    def test_userdefn_eqbox(self, universe, tmpdir):
+    def test_userdefn_eqbox(self, universe, tmpdir, client_DensityAnalysis):
         with warnings.catch_warnings():
             # Do not need to see UserWarning that box is too small
             warnings.simplefilter("ignore")
             self.check_DensityAnalysis(
                 universe.select_atoms(self.selections['static']),
                 self.references['static_defined']['meandensity'],
                 tmpdir=tmpdir,
+                client_DensityAnalysis=client_DensityAnalysis,
                 gridcenter=self.gridcenters['static_defined'],
                 xdim=10.0,
                 ydim=10.0,
                 zdim=10.0,
             )
 
-    def test_userdefn_neqbox(self, universe, tmpdir):
+    def test_userdefn_neqbox(self, universe, tmpdir, client_DensityAnalysis):
         self.check_DensityAnalysis(
             universe.select_atoms(self.selections['static']),
             self.references['static_defined_unequal']['meandensity'],
             tmpdir=tmpdir,
+            client_DensityAnalysis=client_DensityAnalysis,
             gridcenter=self.gridcenters['static_defined'],
             xdim=10.0,
             ydim=15.0,
             zdim=20.0,
         )
 
-    def test_userdefn_boxshape(self, universe):
+    def test_userdefn_boxshape(self, universe, client_DensityAnalysis):
         D = density.DensityAnalysis(
-            universe.select_atoms(self.selections['static']),
-            delta=1.0, xdim=8.0, ydim=12.0, zdim=17.0,
-            gridcenter=self.gridcenters['static_defined']).run()
+            universe.select_atoms(self.selections["static"]),
+            delta=1.0,
+            xdim=8.0,
+            ydim=12.0,
+            zdim=17.0,
+            gridcenter=self.gridcenters["static_defined"],
+        ).run(**client_DensityAnalysis)
         assert D.results.density.grid.shape == (8, 12, 17)
 
-    def test_warn_userdefn_padding(self, universe):
+    def test_warn_userdefn_padding(self, universe, client_DensityAnalysis):
```

Comment:
I am not 100% sure if we need to test warnings for all parallel modalities, too, but it's the safe approach and you've already converted all tests so let's leave it in. Just something to think about in the future if our test run times are too long (again...).